### PR TITLE
layers: Make swizzle less restrictive

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
This is to address https://github.com/KhronosGroup/Vulkan-Portability/issues/27.

### CODE REVIEW

**Comment by Tony-LunarG** [comment]
Is there a spec change coming to clarify what constitutes an identity swizzle?  If not, or if this is going first, I'd add a comment to the 04456 check mentioning that identity could be either value.

**Review by Tony-LunarG — APPROVED** [review]
LGTM with a comment or delay for spec change

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -5827,11 +5827,20 @@ bool CoreChecks::PreCallValidateCreateImageView(VkDevice device, const VkImageVi
...
+            // VK_COMPONENT_SWIZZLE_IDENTITY."
+            //       However, issue https://github.com/KhronosGroup/Vulkan-Portability/issues/27 points out that the identity can
+            //       also be defined via R, G, B, A enums in the correct order. The spec for VUID 04465 should be updated to make
+            //       this clearer.
             if ((VK_FALSE == enabled_features.portability_subset_features.imageViewFormatSwizzle) &&
-                (pCreateInfo->components.r != VK_COMPONENT_SWIZZLE_IDENTITY ||
-                 pCreateInfo->components.g != VK_COMPONENT_SWIZZLE_IDENTITY ||
-                 pCreateInfo->components.b != VK_COMPONENT_SWIZZLE_IDENTITY ||
-                 pCreateInfo->components.a != VK_COMPONENT_SWIZZLE_IDENTITY)) {
+                (!((pCreateInfo->components.r == VK_COMPONENT_SWIZZLE_IDENTITY) ||
```
[the remark this answers is not in the document]
Comment (ncesario-lunarg) [reply layers/buffer_validation.cpp:5836]: Yep, I should use that :)

**Comment by sfricke-samsung** [comment]
in 1.x.185 spec, `VUID-VkImageViewCreateInfo-imageViewFormatSwizzle-04465` was updated to match this PR, so ready to merge IMO

**Pull request merged**